### DISCUSSION TITLE
Low speed performance on a Raspberry Pi 3 Model B Plus

### DISCUSSION THREAD

**Discussion by remmius** [dscripka/openWakeWord#180] [General]
Hi, thanks for the great work!
The detection works very well on my Linux-desktop computer. However, on my Pi 3 with 32-bit OS I only achieve a runtime of ~200ms-250ms per 80ms audio-sample just by using owwModel.predict() with one keyword running on the default .tflite models. As it is a 32bit-arm, i did not manage to install the Onnxruntime (yet..) and hence I deactivated the Onnxruntime dependencies. Does this explain why my performance is so much slower then what is mentioned in the description? Or is there another trick (e.g.64bit OS) to reach the mentioned performance on the Pi3 ?
Thanks !

**Comment by dscripka** [comment]
Hmm, that is interesting. All of my testing is with a 64-bit OS on the RPi3, yes, so that might be the main issue.

In general, I found that the .tflite models are about ~10-20% faster than ONNX on ARM, so in general that is recommended. But perhaps on 32-bit this isn't the case.

**Reply by remmius** [reply]
Thanks for coming back to me. 
Based on your feedback, I did some further investigation and identified a non-sufficient power supply of the RPi as the root cause...did not expect this big impact.
By updating the power supply, I managed to run the .tflite-model in 'real time' without problems. The 32/64bit-question did not have a 'visible' impact on the RPi3. 
I also tried to run it on my RPi2 Bv1.1, but unfortunately it ended up needing ~160-190ms per sample for the prediction. For this i guess I would have to check out more the microWakeWord models. 
Anyway, thanks again for the great lib and models!
